service/service.go

```diff
@@ -345,6 +348,10 @@ func (s *Service) AuthenticateUser(ctx context.Context, req *pbsvc.UserRequest)
 		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
 
+	if auth.PermissionEnumMap[matchedUser.GetPermissionLevel()] < auth.UserRegistration {
+		logger.Error(consts.AuthenticateUserTag, consts.MsgErrGeneratingAuthToken)
+		return nil, status.Error(codes.Unauthenticated, consts.MsgErrGeneratingAuthToken)
+	}
```

Comment:
`AuthenticateUser` only authenticates user with a minimum permission level of `UserRegistration`.